src/Orders.sol

```diff
@@ -5,103 +5,110 @@ import {IERC20} from "openzeppelin-contracts/contracts/token/ERC20/IERC20.sol";
+    ///      outputs are expected to be received on the target chain(s).
+    /// @dev Fees paid to the Builders for fulfilling the Orders
+    ///      can be included within the "exchange rate" between inputs and outputs.
+    /// @dev The Builder claims the inputs from the contract by submitting `sweep` transactions within the same block.
+    /// @dev The Rollup STF MUST NOT apply `initiate` transactions to the rollup state
+    ///      UNLESS the outputs are delivered on the target chains within the same block.
+    /// @param deadline - The deadline by which the Order must be fulfilled.
+    /// @param inputs - The token amounts offered by the swapper in exchange for the outputs.
+    /// @param outputs - The token amounts that must be received on their target chain(s) in order for the Order to be executed.
+    /// @custom:reverts OrderExpired if the deadline has passed.
+    /// @custom:emits Order if the transaction mines.
+    function inititate(uint256 deadline, Input[] calldata inputs, Output[] calldata outputs) external payable {
         // check that the deadline hasn't passed
         if (block.timestamp >= deadline) revert OrderExpired();
 
-        if (tokenIn == address(0)) {
-            require(amountIn == msg.value);
-        } else {
-            IERC20(tokenIn).transferFrom(msg.sender, address(this), amountIn);
-        }
+        // transfer inputs to this contract
+        _transferInputs(inputs);
 
-        // emit the swap event
-        emit Swap(targetChainId, tokenIn, tokenOut, recipient, deadline, amountIn, amountOut);
+        // emit
+        emit Order(deadline, inputs, outputs);
+    }
+
+    /// @notice Transfer the Order inputs to this contract, where they can be collected by the Order filler.
+    function _transferInputs(Input[] calldata inputs) internal {
+        uint256 value = msg.value;
+        for (uint256 i; i < inputs.length; i++) {
+            if (inputs[i].token == address(0)) {
+                // this line should underflow if there's an attempt to spend more ETH than is attached to the transaction
+                value -= inputs[i].amount;
+            } else {
+                IERC20(inputs[i].token).transferFrom(msg.sender, address(this), inputs[i].amount);
+            }
+        }
     }
 
     /// @notice Transfer the entire balance of ERC20 tokens to the recipient.
-    /// @dev Called by the Builder within the same block as users' `swap` transactions
-    ///      to claim the amounts of `tokenIn`.
+    /// @dev Called by the Builder within the same block as users' `initiate` transactions
+    ///      to claim the `inputs`.
     /// @dev Builder MUST ensure that no other account calls `sweep` before them.
-    /// @param token - The token to transfer.
     /// @param recipient - The address to receive the tokens.
-    function sweep(address token, address recipient) public {
+    /// @param token - The token to transfer.
+    function sweep(address recipient, address token) public {
```

Comment:
any reason for switching the order?

Reply:
answer is yes but it's a bit convoluted. i can explain, bear with me. 

I like to keep params in the same order everywhere in a contract. if a struct has params in a certain order, i like function signatures, events, etc. to keep the params in the exact same order too. i think it's good for readability and maintainability. 

previously there was no struct, so i put params in the same order that they're used within the function logic: 
```
IERC20(token).transfer(recipient, amount);
```
so, i made the params
```
(address token, address recipient, uint256 amount)
```

Now, this PR adds in the following struct: 
```
    /// @notice Tokens that must be receive for a valid order fulfillment
    /// @dev From ERC-7683
    struct Output {
        /// @dev The address of the ERC20 token on the destination chain
        /// @dev address(0) used as a sentinel for the native token
        address token;
        /// @dev The amount of the token to be sent
        uint256 amount;
        /// @dev The address to receive the output tokens
        address recipient;
        /// @dev The destination chain for this output
        uint32 chainId;
    }
```
The other is now `(token, amount, recipient)` - this probably makes more sense, to group token & amount together. 

It would be most consistent to have this exact ordering elsewhere in the contract too. However, we have this event: 
```
    event Sweep(address indexed recipient, address indexed token, uint256 amount);
```
Where `recipient` & `token` are `indexed` event params. 

I like to order indexed params first in events if possible, because in the transaction receipt they get pulled up to the top and I think it's nice to try to keep them in the same order if possible. 

So, I could either do: 
```
(address indexed token, uint256 amount, address indexed recipient)
```
^^ A) keeps the same order as the Struct, but puts a non-indexed param before an indexed param 
```
(address indexed recipient, address indexed token, uint256 amount)
```
^^ B) keeps indexed params first, changes the order of the struct
```
(address indexed token, address indexed recipient, uint256 amount)
```
^^ C) keeps indexed params first, changes the order of the struct, and separates token & amount

I went for option B, but I would not die on the hill that it's necessary